route/v1/types.go

```diff
@@ -282,3 +282,10 @@ const (
 	//          Note that this will not match acme.test only *.acme.test
 	WildcardPolicySubdomain WildcardPolicyType = "Subdomain"
 )
+
+// Route Annotations
+const (
+	// OpenShiftAllowNonDNSCompliantHostAnnotation is an instruction to allow non-DNS Compliant
+	// host names in a route configuration
```

Comment:
This would be a good place to elaborate on the semantics, namely that the annotation tells the API to admit the route even if the host fails strict validation, but the router may still reject the route.